### ISSUE TITLE
Log queries are still slow

### ISSUE THREAD

**Issue by et** [highlight/highlight#4623]
Despite #4568, log queries are still slow.

We now have better [tracing](https://app.datadoghq.com/apm/traces?query=operation_name%3Alogs&cols=core_service%2Ccore_resource_name%2Clog_duration%2Clog_http.method%2Clog_http.status_code&historicalData=true&messageDisplay=inline&sort=desc&spanType=all&spanViewType=metadata&start=1678822719334&end=1678995519334&paused=false) and can analyze the queries a bit better.

**Comment by et** [comment]
For loading the next page, we currently don't include the time frame. In theory this should be optional but I have noticed the number of parts to sift through to be less:

```sql
EXPLAIN indexes = 1
SELECT
    Timestamp,
    UUID,
    SeverityText,
    Body,
    LogAttributes,
    TraceId,
    SpanId,
    SecureSessionId
FROM logs
WHERE (ProjectId = 1) AND (toUInt64(toDateTime(Timestamp)) <= 1679072415) AND (toUInt64(toDateTime(Timestamp)) >= 1679062415) AND ((toUInt64(toDateTime(Timestamp)) < 1679072415) OR (UUID < '84bc4147-5919-418f-9183-7a0a59e25a27'))
ORDER BY
    toUnixTimestamp(Timestamp) DESC,
    UUID DESC
LIMIT 101

Query id: 54eff4d1-ff84-4f5c-9999-4304dd62b354

TimeoutSetter: Can't reset timeouts: Poco::Exception. Code: 1000, e.code() = 22, Invalid argument, Stack trace (when copying this message, always include the lines below):

 (version 22.7.2.1)
Connecting to gt95v294x8.us-east-2.aws.clickhouse.cloud:9440 as user default.
Connected to ClickHouse server version 22.13.1 revision 54461.

ClickHouse client version is older than ClickHouse server. It may lack support for new features.

┌─explain────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ Expression (Projection)                                                                                                                                                                                                                                    │
│   Limit (preliminary LIMIT (without OFFSET))                                                                                                                                                                                                               │
│     Sorting (Sorting for ORDER BY)                                                                                                                                                                                                                         │
│       Expression (Before ORDER BY)                                                                                                                                                                                                                         │
│         Filter (WHERE)                                                                                                                                                                                                                                     │
│           ReadFromMergeTree (default.logs)                                                                                                                                                                                                                 │
│           Indexes:                                                                                                                                                                                                                                         │
│             MinMax                                                                                                                                                                                                                                         │
│               Keys:                                                                                                                                                                                                                                        │
│                 Timestamp                                                                                                                                                                                                                                  │
│               Condition: and(and((toUInt64(toDateTime(Timestamp)) in [1679062415, +Inf)), (toUInt64(toDateTime(Timestamp)) in (-Inf, 1679072415])), and((toUInt64(toDateTime(Timestamp)) in [1679062415, +Inf)), (toUInt64(toDateTime(Timestamp)) in (-Inf, 1679072415]))) │
│               Parts: 11/211                                                                                                                                                                                                                                │
│               Granules: 6321/170029                                                                                                                                                                                                                        │
│             Partition                                                                                                                                                                                                                                      │
│               Condition: true                                                                                                                                                                                                                              │
│               Parts: 11/11                                                                                                                                                                                                                                 │
│               Granules: 6321/6321                                                                                                                                                                                                                          │
│             PrimaryKey                                                                                                                                                                                                                                     │
│               Keys:                                                                                                                                                                                                                                        │
│                 ProjectId                                                                                                                                                                                                                                  │
│               Condition: (ProjectId in [1, 1])                                                                                                                                                                                                             │
│               Parts: 9/11                                                                                                                                                                                                                                  │
│               Granules: 2673/6321                                                                                                                                                                                                                          │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

23 rows in set. Elapsed: 0.273 sec.
````

**Comment by et** [comment]
Fixed in #4667
